rlpx.md

```diff
@@ -200,67 +200,47 @@ There are two kinds of connections which can be established. A node can connect
-		 
-    Known: authInitiator = E(remote-pubk, S(ephemeral-privk, token ^ nonce) || H(ephemeral-pubk) || pubk || nonce || 0x1)
-           authRecipient = E(remote-pubk, remote-ephemeral-pubk || nonce || 0x1) // token found
-           authRecipient = E(remote-pubk, remote-ephemeral-pubk || nonce || 0x0) // token not found
-    
-	static-shared-secret = ecdh.agree(privkey, remote-pubk)
-	ephemeral-shared-secret = ecdh.agree(ephemeral-privk, remote-ephemeral-pubk)
-
-Values generated following the handshake (see below for steps):
-
-    ephemeral-shared-secret = ecdh.agree(ephemeral-privkey, remote-ephemeral-pubk)
-    shared-secret = sha3(ephemeral-shared-secret || sha3(nonce || initiator-nonce))
-    token = sha3(shared-secret)
-    aes-secret = sha3(ephemeral-shared-secret || shared-secret)
-    # destroy shared-secret
-    mac-secret = sha3(ephemeral-shared-secret || aes-secret)
-    # destroy ephemeral-shared-secret
-    
-    Initiator:
-    egress-mac = sha3.update(mac-secret ^ recipient-nonce || auth-sent-init)
-    # destroy nonce
-    ingress-mac = sha3.update(mac-secret ^ initiator-nonce || auth-recvd-ack)
-    # destroy remote-nonce
-    
-    Recipient:
-    egress-mac = sha3.update(mac-secret ^ initiator-nonce || auth-sent-ack)
-    # destroy nonce
-    ingress-mac = sha3.update(mac-secret ^ recipient-nonce || auth-recvd-init)
-    # destroy remote-nonce
+## Handshake Protocol
 
-Creating authenticated connection:
+The initiator generates a random key pair and sends the following message:
 
-    1. initiator generates auth from ecdhe-random, static-shared-secret, and nonce (auth = authInitiator handshake)
-    2. initiator connects to remote and sends auth
-	
-    3. optionally, remote decrypts and verifies auth (checks that recovery of signature == H(ephemeral-pubk))
-    4. remote generates authAck from remote-ephemeral-pubk and nonce (authAck = authRecipient handshake)
-	
-	optional: remote derives secrets and preemptively sends protocol-handshake (steps 9,11,8,10)
-	
-    5. initiator receives authAck
-    6. initiator derives shared-secret, aes-secret, mac-secret, ingress-mac, egress-mac
-    7. initiator sends protocol-handshake
-	
-    8. remote receives protocol-handshake
-    9. remote derives shared-secret, aes-secret, mac-secret, ingress-mac, egress-mac
-    10. remote authenticates protocol-handshake
-    11. remote sends protocol-handshake
-	
-    12. initiator receives protocol-handshake
-    13. initiator authenticates protocol-handshake
-    13. cryptographic handshake is complete if mac of protocol-handshakes are valid; permanent-token is replaced with token
-    14. begin sending/receiving data
-	
-	All packets following auth, including protocol negotiation handshake, are framed.
+    rlpx-version = 5 (encoded as single byte)
+    auth-msg-initiator = ecies.encrypt(recipient-pubk,
+        initiator-pubk || initiator-ephemeral-pubk || initiator-nonce || rlpx-version)
```

Comment:
The signature and token indicator need to be kept. They are important for authenticity in the first two steps of the handshake. Unless we move directly to OTR, which is possible, however until then its removal obviates the recipients ability to *quickly* verify the initiator as a trusted party.